src/pages/playlists/[slug].tsx

```diff
@@ -13,16 +13,18 @@ type PlaylistProps = {
 
 const Playlist: FunctionComponent<PlaylistProps> = ({playlist}) => {
   const initialData = playlist
-  const {data} = useSWR(playlist.url, fetcher, {initialData})
+  const {data} = useSWR(playlist.url, fetcher)
```

Comment:
this "initial data" call made it work very strangely and links to other playlists would load the first playlist. Removing this and combining them like below fixed it.